tests/AppTest.php

```diff
@@ -315,6 +316,31 @@ public function testCanAddAndExecuteHooks()
         $this->assertEquals('x-def', $result);
     }
 
+    public function testAllowRouteOverrides()
+    {
+        App::setAllowOverride(false);
+        $this->assertFalse(App::getAllowOverride());
+        App::get('/')->action(function () {
+            echo 'Hello first';
+        });
+
+        $this->expectException(Exception::class);
+        App::get('/')->action(function () {
+            echo 'Hello second';
+        });
+
+        // Test success
+        App::setAllowOverride(true);
+        $this->assertTrue(App::getAllowOverride());
+        App::get('/')->action(function () {
+            echo 'Hello first';
+        });
+
+        App::get('/')->action(function () {
+            echo 'Hello second';
+        });
```

Comment:
This will never run because an uncaught exception will be thrown (`expectException` does not catch)

You need to do something like this instead:

```php
        try {
            App::get('/')->action(function () {
                echo 'Hello second';
            });
            $this->fail('Failed to throw exception');
        } catch (Exception $e) {
            // Threw exception as expected
        }
```